java/src/main/java/com/powsybl/python/security/SecurityAnalysisCFunctions.java

```diff
@@ -138,14 +138,31 @@ public static void addContingency(IsolateThread thread, ObjectHandle contingency
         });
     }
 
-    private static LoadFlowResult.ComponentResult.Status getStatus(LimitViolationsResult result) {
-        return result.isComputationOk() ? LoadFlowResult.ComponentResult.Status.CONVERGED : LoadFlowResult.ComponentResult.Status.FAILED;
+    private static void setPostContingencyResultInSecurityAnalysisResultPointer(PyPowsyblApiHeader.PostContingencyResultPointer contingencyPtr, PostContingencyResult postContingencyResult) {
+        contingencyPtr.setContingencyId(CTypeUtil.toCharPtr(postContingencyResult.getContingency().getId()));
+        contingencyPtr.setStatus(postContingencyResult.getStatus().ordinal());
+        List<LimitViolation> limitViolations = postContingencyResult.getLimitViolationsResult().getLimitViolations();
```

Comment:
Maybe we could create a function to share limit violations code between pre and post contingency?